api/v1alpha1/webhook_suite_test.go

```diff
@@ -126,7 +129,338 @@ var _ = BeforeSuite(func() {
+					Server:                      "https://10.51.140.43:8443/era/v0.9",
+				},
+				Instance: Instance{
+					CredentialSecret:     "db-instance-secret",
+					DatabaseInstanceName: "db-instance-name",
+					Type:                 "postgres",
+					Size:                 8,
+					TimeZone:             "UTC",
+				},
+			},
+		}
+		err := k8sClient.Create(context.Background(), database)
+		Expect(err).To(HaveOccurred())
+
+		// Extract the error message from the error object
+		errMsg := err.(*errors.StatusError).ErrStatus.Message
+		Expect(errMsg).To(ContainSubstring("Initial Database size must be specified with a value 10 GBs or more"))
+
+	})
+})
+
+var _ = Describe("Webhooks: Profiles missing (Open-source)", func() {
+	It("Testing webhooks", func() {
+
+		database := &Database{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "db",
+				Namespace: "default",
+			},
+			Spec: DatabaseSpec{
+				NDB: NDB{
+					ClusterId:                   "27bcce67-7b83-42c2-a3fe-88154425c170",
+					SkipCertificateVerification: true,
+					CredentialSecret:            "ndb-secret",
+					Server:                      "https://10.51.140.43:8443/era/v0.9",
+				},
+				Instance: Instance{
+					CredentialSecret:     "db-instance-secret",
+					DatabaseInstanceName: "db-instance-name",
+					Size:                 10,
+					TimeZone:             "UTC",
+					Type:                 "postgres",
+				},
+			},
+		}
+		err := k8sClient.Create(context.Background(), database)
+		fmt.Print(err)
+		Expect(err).ToNot(HaveOccurred())
+	})
+})
+
+var _ = Describe("Webhooks: Profiles missing (Closed-source)", func() {
+	It("Testing webhooks", func() {
+
+		database := &Database{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "db",
+				Namespace: "default",
+			},
+			Spec: DatabaseSpec{
+				NDB: NDB{
+					ClusterId:                   "27bcce67-7b83-42c2-a3fe-88154425c170",
+					SkipCertificateVerification: true,
+					CredentialSecret:            "ndb-secret",
+					Server:                      "https://10.51.140.43:8443/era/v0.9",
+				},
+				Instance: Instance{
+					CredentialSecret:     "db-instance-secret",
+					DatabaseInstanceName: "db-instance-name",
+					Size:                 10,
+					TimeZone:             "UTC",
+					Type:                 "mssql",
+				},
+			},
+		}
+		err := k8sClient.Create(context.Background(), database)
+		fmt.Print(err)
+		Expect(err).To(HaveOccurred())
+
+		// Extract the error message from the error object
+		errMsg := err.(*errors.StatusError).ErrStatus.Message
+		Expect(errMsg).To(ContainSubstring("Software Profile must be provided for the closed-source database engines"))
+	})
+})
+
+var _ = Describe("Webhooks: Time Machine missing (Open-source)", func() {
+	It("Testing webhooks", func() {
+
+		database := &Database{
+			ObjectMeta: metav1.ObjectMeta{
+				// need to provide a unique name to avoid getting " "db" already exists"
+				// this is because the previous test was a happy case and a db CR was successfully created
```

Comment:
Hey @shenoypritika Good suggestion. Just noticed that 8 of these 10 tests are NOT actually creating any resource as we are performing the negative testing i.e. assert the error returned by the Admission Webhooks. In the case of reject from the Webhooks, K8s will likely not create any resource so nothing to be cleaned up with 'AfterEach()'.

We do have 2 tests that successfully create a resource and those are being deleted by the AfterSuite(), so I am not sure running AfterEach() after ever step would help here, in fact it may make tests run slower. Generally we use Before/AfterEach() if we need isolation, but we do not have any such requirement here.

But I am ok to add AfterEach() if you can see a strong reason otherwise. Let me know.

Reply:
I think adding AfterEach will make the test suite more extensible. In the future, if we add more we need not worry about resource name conflicts.

Reply:
Sure, as discussed the cleanup after every test makes the test suite slower to run, but I am ok with it and have no specific preference on that. Updated the code. Resolving the comment.